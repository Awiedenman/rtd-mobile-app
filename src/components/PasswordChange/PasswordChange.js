import React, { Component } from 'react';

import { auth } from '../../firebase';

const INITIAL_STATE = {
  passwordOne: '',
  passwordTwo: '',
  error: null
};

export class PasswordChangeForm extends Component {
  constructor(props) {
    super(props);
    this.state = { ...INITIAL_STATE };
  }

  onSubmit = (event) => {
    const { passwordOne } = this.state;

    auth.doPasswordUpdate(passwordOne)
      .then(() => {
        this.setState({ ...INITIAL_STATE });
      })
      .catch(error => {
        this.setState({error: error});
      });

    event.preventDefault();
  }

  handleChange = (event) => {
    const { name, value } = event.target;
    this.setState({
      [name]: value
    });
  }

  render() {
    const {
      passwordOne,
      passwordTwo,
      error
    } = this.state;

    const isInvalid =
      passwordOne !== passwordTwo ||
      passwordOne === '';

    return (
      <form onSubmit={this.onSubmit}>
        <input
          className='password-one'
          name="passwordOne"
          value={passwordOne}
          onChange={this.handleChange}
          type="password"
          placeholder="New Password"
        />
        <input
          className='password-two'
          name="passwordTwo"
          value={passwordTwo}
          onChange={this.handleChange}
          type="password"
          placeholder="Confirm New Password"
        />
        <button disabled={isInvalid} type="submit">
          Reset My Password
        </button>

        { error && <p>{error.message}</p> }
      </form>
    );
  }
}

export default PasswordChangeForm;