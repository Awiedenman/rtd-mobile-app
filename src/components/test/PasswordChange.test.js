import React from 'react';
import { PasswordChangeForm } from '../PasswordChange/PasswordChange';
import { shallow } from 'enzyme';

describe('PasswordChange', () => {
  it('should matchSnapshot', () => {
    const wrapper = shallow(<PasswordChangeForm/>);

    expect(wrapper).toMatchSnapshot();
  });

  describe('handleChange', () => {
    it('should call handleChange of change of the input field ', () => {
      const wrapper = shallow(<PasswordChangeForm/>);
      const mockPassword = '123456';
      const handleChange = jest.fn();
      const event = {
        // preventDefault() {},
        target: {
          name: 'passwordOne',
          value: '123456'
        }
      };
      
      handleChange(event);

      wrapper.find('.password-one').simulate('change', event);
      expect(handleChange).toHaveBeenCalled();
      expect(wrapper.state('passwordOne')).toEqual(mockPassword);
    });
  });

  describe('handleSubmit', () => {
    it('should call handlesubmit of change of the input field ', () => {
      const wrapper = shallow(<PasswordChangeForm/>);
      const onSubmit = jest.fn();
      const event = {
        preventDefault() {},
        target: {
          value: ''
        }
      };
      
      onSubmit(event);
      wrapper.find('form').simulate('sumbit', event);
      expect(onSubmit).toHaveBeenCalled();
    });
  });
});
