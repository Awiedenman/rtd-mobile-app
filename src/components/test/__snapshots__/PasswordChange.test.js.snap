// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordChange should matchSnapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PasswordChangeForm />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          className="password-one"
          name="passwordOne"
          onChange={[Function]}
          placeholder="New Password"
          type="password"
          value=""
/>,
        <input
          className="password-two"
          name="passwordTwo"
          onChange={[Function]}
          placeholder="Confirm New Password"
          type="password"
          value=""
/>,
        <button
          disabled={true}
          type="submit"
>
          Reset My Password
</button>,
        null,
      ],
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "password-one",
          "name": "passwordOne",
          "onChange": [Function],
          "placeholder": "New Password",
          "type": "password",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "password-two",
          "name": "passwordTwo",
          "onChange": [Function],
          "placeholder": "Confirm New Password",
          "type": "password",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Reset My Password",
          "disabled": true,
          "type": "submit",
        },
        "ref": null,
        "rendered": "Reset My Password",
        "type": "button",
      },
      null,
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            className="password-one"
            name="passwordOne"
            onChange={[Function]}
            placeholder="New Password"
            type="password"
            value=""
/>,
          <input
            className="password-two"
            name="passwordTwo"
            onChange={[Function]}
            placeholder="Confirm New Password"
            type="password"
            value=""
/>,
          <button
            disabled={true}
            type="submit"
>
            Reset My Password
</button>,
          null,
        ],
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "password-one",
            "name": "passwordOne",
            "onChange": [Function],
            "placeholder": "New Password",
            "type": "password",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "password-two",
            "name": "passwordTwo",
            "onChange": [Function],
            "placeholder": "Confirm New Password",
            "type": "password",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Reset My Password",
            "disabled": true,
            "type": "submit",
          },
          "ref": null,
          "rendered": "Reset My Password",
          "type": "button",
        },
        null,
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
